src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -2018,6 +2018,214 @@ TEST_F(
   EXPECT_EQ(1u, app_list.size());
 }
 
+TEST_F(
+    ApplicationManagerImplTest,
+    RegisterApplication_WEPNonMediaAppRegisterActivateDeactivateExit_EXPECT_CORRECT_HMI_STATES) {
+  EXPECT_CALL(
+      mock_session_observer_,
+      GetDataOnSessionKey(kConnectionKey,
+                          _,
+                          _,
+                          testing::An<connection_handler::DeviceHandle*>()))
+      .WillOnce(DoAll(SetArgPointee<3u>(kDeviceId), Return(0)));
+
+  smart_objects::SmartObject rai(smart_objects::SmartType_Map);
+  smart_objects::SmartObject& params = rai[strings::msg_params];
+  params[strings::app_id] = kAppId;
+  params[strings::language_desired] = mobile_api::Language::EN_US;
+  params[strings::hmi_display_language_desired] = mobile_api::Language::EN_US;
+
+  rai[strings::params][strings::connection_key] = kConnectionKey;
+  rai[strings::msg_params][strings::app_name] = kAppName;
+  rai[strings::msg_params][strings::app_hmi_type] = "WEB_VIEW";
+  rai[strings::msg_params][strings::is_media_application] = false;
+  rai[strings::msg_params][strings::sync_msg_version][strings::minor_version] =
+      APIVersion::kAPIV2;
+  rai[strings::msg_params][strings::sync_msg_version][strings::major_version] =
+      APIVersion::kAPIV6;
+
+  rai[strings::params][strings::protocol_version] =
+      protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_2;
+
+  const auto rai_ptr = std::make_shared<smart_objects::SmartObject>(rai);
+
+  std::unique_ptr<plugin_manager::RPCPluginManager> rpc_plugin_manager(
+      new MockRPCPluginManager());
+  app_manager_impl_->SetPluginManager(rpc_plugin_manager);
+  auto wep_app = app_manager_impl_->RegisterApplication(rai_ptr);
+  wep_app->set_is_media_application(false);
+
+  EXPECT_EQ(protocol_handler::MajorProtocolVersion::PROTOCOL_VERSION_2,
+            wep_app->protocol_version());
+  EXPECT_EQ(APIVersion::kAPIV2, wep_app->version().min_supported_api_version);
+  EXPECT_EQ(APIVersion::kAPIV6, wep_app->version().max_supported_api_version);
+
+  const WindowID main_window_id = 0;
```

Comment:
```suggestion
  const WindowID main_window_id = mobile_apis::PredefinedWindows::DEFAULT_WINDOW;
```

Reply:
@AKalinich-Luxoft Thank you. I found a better way) See 9a4d4bb83b71d728d39c231e38bca1bb680fb997